plugins/gcov/gcov.rake

```diff
@@ -154,67 +154,268 @@ if PROJECT_USE_DEEP_DEPENDENCIES
 end
 
 namespace UTILS_SYM do
-  def gcov_args_builder(opts)
+  # Build the gcovr report generation common arguments.
+  def gcovr_args_builder_common(opts)
     args = ""
-    args += "-r \"#{opts[:gcov_report_root] || '.'}\" "
-    args += "-f \"#{opts[:gcov_report_include]}\" " unless opts[:gcov_report_include].nil?
-    args += "-e \"#{opts[:gcov_report_exclude] || GCOV_FILTER_EXCLUDE}\" "
-    [ :gcov_fail_under_line, :gcov_fail_under_branch, :gcov_html_medium_threshold, :gcov_html_high_threshold].each do |opt|
+    args += "--root \"#{opts[:gcov_report_root] || '.'}\" "
+    args += "--config \"#{opts[:gcov_config_file]}\" " unless opts[:gcov_config_file].nil?
+    args += "--filter \"#{opts[:gcov_report_include]}\" " unless opts[:gcov_report_include].nil?
+    args += "--exclude \"#{opts[:gcov_report_exclude] || GCOV_FILTER_EXCLUDE}\" "
```

Comment:
@aschlicht I know you are working on changes for `:report_types:` list. I'm looking forward to see this new version :).

I just have a question about these options and description in README.md

```
# Keep only source files that match this filter. (gcovr --filter).
:report_include: "^src"
 
# Exclude source files that match this filter. (gcovr --exclude).
:report_exclude: "^build|^vendor|^test|^support"
```
`GCOV_FILTER_EXCLUDE` has value `'^vendor.*|^build.*|^test.*|^lib.*'` and is always ORed with `gcov_report_exclude`

`gcov_report_include` is not ORed with any settings but in the README.md you mentioned about filter for `^src`

Settings mentioned in the README.md are related to the default behaviour or it is just an example how to use these options?

Reply:
@CezaryGapinski I was treating the settings in the README.md as examples and documenting some defaults in the comments above the settings. I can update the `:report_exclude:` in README.md to show the default.

Regarding the OR, I thought the `||` was a logical OR. So if project.yml options specify `:report_exclude:`, then use the settings in project.yml. If the settings are not included in the options, then use the default specified by `GCOV_FILTER_EXCLUDE`. I read about this Ruby syntax on the internet. Did I interpret it incorrectly?

I have some of the `:report_types:` version implemented. Shall we support both GcovrHtmlBasic and GcovrHtmlDetailed? We could output only detailed in that case.

Reply:
> Regarding the OR, I thought the || was a logical OR. So if project.yml options specify :report_exclude:, then use the settings in project.yml. If the settings are not included in the options, then use the default specified by GCOV_FILTER_EXCLUDE. I read about this Ruby syntax on the internet. Did I interpret it incorrectly?

Now I understand it is clear for me now. I guess you have right about || operator.
I just was not sure about difference between `'^vendor.*|^build.*|^test.*|^lib.*'` and `"^build|^vendor|^test|^support"`.

Reply:
Thanks for confirming the use of the || operator.

I changed the default in the README.md as you suggested.